### ISSUE TITLE
Error when compiling test suite for CUDA on Yosemite

### ISSUE THREAD

**Issue by YuMan-Tam** [arrayfire/arrayfire#197]
There seems to be a linking error when `cmake` was used to compile arrayfire on OS X Yosemite with CUDA 6.5. The error only happens when `BUILD_TEST` is on and only in the file `fft_cuda`. I suspect some CUDA libraries are missing. But, in the corresponding `link.txt` file,  l saw plenty of libraries with `.../cuda/lib/...`. 

```
Linking CXX executable diff1_cuda
[ 96%] Built target diff1_cuda
Linking CXX executable fft_cuda
Undefined symbols for architecture x86_64:
  "af::array::array<std::__1::complex<float> >(af::dim4 const&, std::__1::complex<float> const*, af::af_source_t, int)", referenced from:
      void cppFFTTest<std::__1::complex<float>, std::__1::complex<float>, false>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int, int, int) in fft.cpp.o
      void cppFFTTest<std::__1::complex<float>, std::__1::complex<float>, true>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int, int, int) in fft.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [test/fft_cuda] Error 1
make[1]: *** [test/CMakeFiles/fft_cuda.dir/all] Error 2
Linking CXX executable diff2_cuda
[ 96%] Built target diff2_cuda
Linking CXX executable convolve_cuda
[ 96%] Built target convolve_cuda
make: *** [all] Error 2
```

**Comment by pavanky** [comment]
This issue should be fixed in latest devel.

**Comment by dvasiliu** [comment]
Well, I tried the latest devel (as of March 6th) and the same error shows up.

**Comment by pavanky** [comment]
@dvasiliu Can you try running the following command 

```
xcode-select --install
```

and try to compile again ?

**Comment by dvasiliu** [comment]
Hi Pavan,

Thanks for getting some information. I did what you suggested and this is what I get:

xcode-select: error: command line tools are already installed, use "Software Update" to install updates

**Comment by pavanky** [comment]
@dvasiliu #472 may fix the issue once it is merged into devel. Can you try to use the following to test if it works on your machine ?

```
git clone https://github.com/umar456/arrayfire.git -b arrayfire_clang
cd arrayfire_clang
// follow the build instructions from the wiki
```

**Comment by dvasiliu** [comment]
It works !!  Yay ! Many thanks !

**Comment by pavanky** [comment]
@dvasiliu The pull request has been merged. You can use the mainline arrayfire without any issues from now.
